Angular/src/app/app.component.ts

```diff
@@ -0,0 +1,98 @@
+import { Component, ViewChild } from '@angular/core';
+import { DxDataGridComponent } from 'devextreme-angular';
+
+import validationEngine from 'devextreme/ui/validation_engine';
+
+import ArrayStore from 'devextreme/data/array_store';
+import { Employee, Service } from './app.service';
+
+@Component({
+    selector: 'app-root',
+    templateUrl: './app.component.html',
+    styleUrls: ['./app.component.css'],
+    providers: [Employee, Service]
+})
+
+export class AppComponent {
+    @ViewChild("grid", {static: false}) grid: DxDataGridComponent;
+
+    formData: Object = {};
+    isNewRecord: boolean = true;
+    visible: boolean = false;
+
+    validationGroupName: string;
+
+    employees: Employee[];
+    employeeStore: ArrayStore;
+
+    positionEditorOptions: Object;
+    notesEditorOptions: Object;
+    phoneEditorOptions: Object;
+    labelTemplates: Object;
+
+    constructor(service: Service) {
+        this.employees = service.getEmployees();
+
+        this.employeeStore = new ArrayStore({
+            data: this.employees,
+            key: "ID",
+        });
+
+        this.positionEditorOptions = { items: service.getPositions(), searchEnabled: true };
+        this.notesEditorOptions = { height: 90, maxLength: 200 };
+        this.phoneEditorOptions = { 
+            mask: "+1 (X00) 000-0000", 
+            maskRules: { X: /[02-9]/ },
+            maskInvalidMessage: "The phone must have a correct USA phone format", 
+        };
+
+        this.labelTemplates = [
+            { name: "name", icon: "dx-icon-user" },
+            { name: "position", icon: "dx-icon-info" },
+            { name: "date", icon: "dx-icon-event" },
+            { name: "address", icon: "dx-icon-home" },
+            { name: "phone", icon: "dx-icon-tel" },
+            { name: "email", icon: "dx-icon-email" },
+        ];
+
+        this.validationGroupName = "gridForm";
+
+        this.addRow = this.addRow.bind(this);
+        this.editRow = this.editRow.bind(this);
+        this.confirmChanges = this.confirmChanges.bind(this);
+        this.hidePopup = this.hidePopup.bind(this);
```

Comment:
Instead of using  bind you can rewrite hidePopup and the same methods using it as a class field instead of a method:

```
    hidePopup = () => { 
        this.visible = false;
    };
```